### PR TITLE
Implement type sploring

### PR DESCRIPTION
**Pull request by kalman**
Display names are relatively similar to `noms show`.
I've also changed the display names for struct values to be more similar
to `noms show`.

### CODE REVIEW

File: cmd/noms/splore/noms_splore.go [hunk 1]
```diff
@@ -191,25 +191,47 @@ func getNodeChildren(v types.Value, parentPath string) (children []nodeChild) {
...
+			i := 0
+			d.IterFields(func(name string, t *types.Type, optional bool) {
+				children[i] = nodeChild{
+					Label: name,
+					Value: info(t, childPath(".%s", name)),
+				}
+				i++
+			})
+		default:
+			children = []nodeChild{} // TODO: cycles?
```
Comment (kalman) [inline cmd/noms/splore/noms_splore.go]: I would test this but the place I usually get cycles - from the photo data - is no longer compatible with this noms version. Will fix later.

Comment (arv) [reply cmd/noms/splore/noms_splore.go]: Cycles are represented as pointer cycles in memory so you would keep getting back to the same struct type over and over.

Comment (arv) [reply cmd/noms/splore/noms_splore.go]: It is easy enough to create a cyclic type programmatically of you want to test this.

Comment (kalman) [reply cmd/noms/splore/noms_splore.go]: > you would keep getting back to the same struct 

Yep I see this with Commit. It's not a big deal. I think the graph library we use would get confused trying to draw a cyclic graph, so shrug

> want to test this 

Yeah, I'm just not sure what I want the behavior to be yet, I guess.

**Pull request merged**